R/zchunk_L270.limits.R

```diff
@@ -0,0 +1,294 @@
+#' module_energy_L270.limits
+#'
+#' Generate GCAM policy constraints which limit model behavior in some way.  In
+#' particular limit the fraction of liquid feedstocks and inputs to electricity
+#' generation which can come from sources other than crude oil.  Constraint the
+#' total amount of subsidy as a fraction of GDP which an economy is will to give
+#' to have net negative emissions.
+#'
+#' @param command API command to execute
+#' @param ... other optional parameters, depending on command
+#' @return Depends on \code{command}: either a vector of required inputs,
+#' a vector of output names, or (if \code{command} is "MAKE") all
+#' the generated outputs: \code{L270.CreditOutput}, \code{L270.CreditInput_elec}, \code{L270.CreditInput_feedstocks}, \code{L270.CreditMkt}, \code{L270.CTaxInput}, \code{L270.NegEmissFinalDemand}, \code{L270.NegEmissFinalDemand_SPA}, \code{L270.NegEmissBudgetMaxPrice}, \code{paste0( "L270.NegEmissBudget_", c("GCAM3", paste0("SSP", 1:5), paste0("gSSP", 1:5), c("spa1", "spa23", "spa4", "spa5")) )}. The corresponding file in the
+#' original data system was \code{L270.limits.R} (energy level2).
+#' @details Generate GCAM policy constraints which enforce limits to liquid feedstocks
+#' and the amount of subsidies given for net negative emissions.
+#' @importFrom assertthat assert_that
+#' @importFrom dplyr filter mutate select
+#' @importFrom tidyr gather spread
+#' @importFrom magrittr %<>%
+#' @author PLP March 2018
+module_energy_L270.limits <- function(command, ...) {
+  if(command == driver.DECLARE_INPUTS) {
+    return(c(FILE = "common/GCAM_region_names",
+             FILE = "energy/A23.globaltech_eff",
+             "L102.gdp_mil90usd_GCAM3_R_Y",
+             "L102.gdp_mil90usd_Scen_R_Y",
+             "L221.GlobalTechCoef_en"))
+  } else if(command == driver.DECLARE_OUTPUTS) {
+    return(c("L270.CreditOutput",
+             "L270.CreditInput_elec",
+             "L270.CreditInput_feedstocks",
+             "L270.CreditMkt",
+             "L270.CTaxInput",
+             "L270.NegEmissFinalDemand",
+             "L270.NegEmissFinalDemand_SPA",
+             "L270.NegEmissBudgetMaxPrice",
+             # TODO: might just be easier to keep the scenarios in a single
+             # table here and split when making XMLs but to match the old
+             # data system we will split here
+             paste0( "L270.NegEmissBudget_", c("GCAM3", paste0("SSP", 1:5), paste0("gSSP", 1:5), c("spa1", "spa23", "spa4", "spa5")) )))
+  } else if(command == driver.MAKE) {
+
+    # Chunk specific constants
```

Comment:
I know that these are only used in this chunk, but wouldn't it be easier to find/change constants if they were all in the constants.R file?